release/models/system/openconfig-system-grpc-tunnel.yang

```diff
@@ -0,0 +1,348 @@
+module openconfig-system-grpc-tunnel {
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/system/grpc-tunnel";
+  prefix "oc-sys-grpc-tun";
+
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-system { prefix oc-sys; }
+  import openconfig-network-instance { prefix oc-netinst; }
+  import openconfig-inet-types { prefix oc-inet; }
+
+  organization
+    "OpenConfig working group";
+  contact
+    "www.openconfig.net";
+
+  description
+    "This module adds configuration and operational state relating to
+    gRPC tunnel (https://github.com/openconfig/grpctunnel/) to invoke
+    one or more dialout TCP over gRPC tunnels to bypass situations in
+    which direct dialin gRPC communications are not feasible.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision 2022-07-08 {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  typedef target-type {
+    type enumeration {
+      enum UNKNOWN {
+        value 0;
+        description
+          "An Unknown/Invalid target type.";
+      }
+      enum SSH {
+        value 22;
+        description
+          "SSH protocol target type.";
+      }
+      enum NETCONF_SSH {
+        value 830;
+        description
+          "NETCONF over SSH target type.";
+      }
+      enum OPENFLOW {
+        value 6653;
+        description
+          "OpenFlow protocol target type.";
+      }
+      enum GNMI_GNOI {
+        value 9339;
+        description
+          "gNMI/gNOI target type.";
+      }
+      enum GRIBI {
+        value 9340;
+        description
+          "gRIBI target type.";
+      }
+      enum P4RT {
+        value 9559;
+        description
+          "P4 Runtime target type.";
+      }
+    }
+    description
+      "Sets the TCP based protocols to be tunneled over the gRPC
+      tunnel.";
+    reference
+      "OpenConfig gRPC Tunnel:
+      https://github.com/openconfig/grpctunnel/blob/master/proto/tunnel/tunnel.proto";
+  }
+
+  grouping tunnel-server-config {
```

Comment:
This seems to be a parallel definition of what was created in proto, here.
https://github.com/openconfig/grpctunnel/blob/master/cmd/target/proto/config/target_config.proto

Do we want to have per-leaf, redudant configuration definitions between proto and YANG?